website/blog/2024-03-18-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,201 @@
+
+
+# Visualization and Sharing Capabilities on MLflow
+
+One of the most powerful features of MLFlow is its [newly introduced visualization capabilities](https://mlflow.org/blog/deep-learning-part-1). These tools enable users to make comparisons across different runs and artifacts. Such visualizations can be integrated into dashboards, facilitating easy sharing with both internal and external stakeholders. Additionally, the centralized storage of metadata allows for effective analysis of performance metrics, like identifying minima and maxima in model performance.
+
+![](mlflow_run_details.png)
+
+Run Details
+
+![](deep_learning_charts_view.png)
+
+Visualizing and comparing curve with MLflow and grouping charts
+
+![](deep_learning_logged_model.png)
+
+Autologged checkpoints in MLflow for flexible deployment
+
+![](mlflow_system_metric.png)
+
+System metrics with MLflow
+
+
+# When to stop training?
+
+For the sake of efficient training and model performance at the chosen task, it is critical to ensure that the model does not overfit to the training data and loses the ability to generalize to unseen examples in the real world scenarios. The easiest way to do this is by tracking the loss metrics for training and validation metrics. Thanks to our MLFlow callback, we log these values and we are in a position to visualize them near-real time. If the validation loss curve isn’t improving (i.e. the loss curve isn’t going down) it’s time to stop the training.  
+
+
+# Evaluate checkpoints of Fine Tuned Model with MLflow
+
+With your training process meticulously tracked and logged by MLflow, you have the flexibility to retrieve and test your model at any checkpoint. To do this, simply employ [mlflow.pyfunc.load\_model()](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=load_model#mlflow.pyfunc.load_model) to load the model from a specific run, and then use the[ .predict()](https://mlflow.org/docs/latest/python_api/mlflow.models.html?highlight=predict#mlflow.models.predict) method for evaluation.
+
+In our example, we focus on the model checkpoint from the 900th epoch. We'll demonstrate how to test this model using a specific prompt and set parameters.
+
+```python
+import mlflow
+
+loaded_model = mlflow.pyfunc.load_model(f"runs:/4b32ccd588794530bd66fd3557c4d42c/checkpoint-900")
+
+# Make a prediction using the loaded model
+loaded_model.predict(
+    {"prompt": "check out Gaucho , near convent garden cost $50 for 2, nice columbian dishes",
+     "temperature": 0.4,
+     "max_tokens": 128,
+    }
+)
+```
```

Comment:
I recommend showing the result, and also a before/after—how did the model respond to an identical prompt prior to fine-tuning?

Reply:
Good idea